src/SupraLib/InputOutput/MetaImageOutputDevice.cpp

```diff
@@ -45,6 +45,7 @@ namespace supra
 		m_valueRangeDictionary.set<string>("filename", "output", "Filename");
 		m_valueRangeDictionary.set<bool>("createSequences", { false, true }, true, "Sequences");
 		m_valueRangeDictionary.set<bool>("active", { false, true }, true, "Active");
+		m_valueRangeDictionary.set<uint32_t>("maxFrames", 1, 99999, 10000, "Maximum frames to be written");
```

Comment:
Did you check whether Slicer / ITK / SimpleITK can handle this?
Originally this was restricted because I only had observed 4 digit frame numbers

Reply:
https://github.com/Kitware/MetaIO/blob/master/src/metaImage.h
line 318: int m_DimSize[10];
---
Given the implementation in this upstream repo (used in ITK and Slicer), the correct limit would be to follow these guidelines, i.e. support maximum 10 dimensions with the limit of a normal integer value. I did not check the specific limits any of the sw-specific implementations might have.